### PR TITLE
[@types/mongoose] add useUnifiedTopology options

### PR DESCRIPTION
**Pull request by various89**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://mongoosejs.com/docs/connections.html#options
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

File: types/mongoose/index.d.ts [hunk 1]
```diff
@@ -512,6 +512,20 @@ declare module "mongoose" {
...
+     */
+    serverSelectionTimeoutMS?: number;
+    /**
+     * With useUnifiedTopology, the MongoDB driver sends a heartbeat every heartbeatFrequencyMS to check on the status of the connection.
+     * A heartbeat is subject to serverSelectionTimeoutMS, so the MongoDB driver will retry failed heartbeats for up to 30 seconds by default.
+     * Mongoose only emits a 'disconnected' event after a heartbeat has failed,
+     * so you may want to decrease this setting to reduce the time between when your server goes down and when Mongoose emits 'disconnected'.
+     * We recommend you do not set this setting below 1000, too many heartbeats can lead to performance degradation.
+     */
+    heartbeatFrequencyMS?:number;
```
Comment (andreialecu) [inline types/mongoose/index.d.ts:528]: nitpick: missing space after `:`

**Comment by danger-public** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## mongoose ([<kbd>unpkg</kbd>](https://unpkg.com/browse/mongoose@latest/))
was missing the following properties:
1. modelSchemas
1. options
1. cast
1. driver
1. deleteModel

<details>
<summary>as well as these 7 other properties...</summary>
<p>ObjectId, Decimal128, Mixed, Number, now, CastError, SchemaTypeOptions</p>
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 8dbfaec92f88fc7c3c4b0b0457a804f6f60c9fe7
</p>

**Review by andreialecu — APPROVED** [review]

**Comment by various89** [comment]
Ready to merge

**Comment by ToucheSir** [comment]
@various89  is there a reason this exists given that `unifiedTopology` and the unified topology options already exist on https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/mongodb/index.d.ts#L559? `mongoose.ConnectionOptions` should be inheriting those by default, so I'm unsure as to why they've been duplicated wholesale here...

**Pull request merged**